.github/workflows/ci_checks.yml

```diff
@@ -142,6 +157,13 @@ jobs:
           disable-animations: true
           script: android-app/gradlew -p android-app --build-cache --no-daemon connectedQaDebugAndroidTest
 
+      - name: Teardown Heroku instance
+        if: always() && steps.start-simple-server.outcome != 'success'
```

Comment:
What does `always` do? 🤔

Reply:
`always()` tells GitHub Actions to always run a step regardless of the outcome of the previous step. However, if you want to conditionally execute a step (without taking the outcome of the previous step into consideration), unless you `AND` the condition with `always()`, GitHub Actions will still ignore checking the condition if the previous step failed. So we need this here.